packages/ui/src/hooks/validation/use-validate-token-amount.ts

```diff
@@ -49,11 +64,12 @@ export const validate = ({
     const decimalIndex = value.indexOf(".");
     const decimal =
       decimalIndex !== -1 ? value.substring(decimalIndex + 1) : undefined;
-    res.decimals = decimals
-      ? decimal
-        ? decimal.length <= decimals
-        : true
-      : true;
+    res.decimals =
```

Comment:
Can we refactor out the nested ternary? `decimal` uses a ternary too, so this is 3 layers deep

Reply:
I kinda like the IIFEs for this